### ISSUE TITLE
Analyzers stalling with failed simulations

### ISSUE THREAD

**Issue by krosenfeld-IDM** [InstituteforDiseaseModeling/idmtools#1674]
**Table of Contents**

    - [Description](#description)
    - [What I Did](#what-i-did)
- [Extras](#extras)

### Description
I find that running an SSMT analyzer stalls if any of the simulations failed due to e.g. node failure.  I deleted the failed item but still do not get any progression of the analyzer.  Is there a way to gracefully handle - maybe ignore - simulations that have failed?

### What I Did

Here is an example analyzer that you can run:

```
from idmtools.core.platform_factory import Platform
from idmtools.entities import IAnalyzer
from idmtools.analysis.platform_anaylsis import PlatformAnalysis

from typing import Dict, Any, Union
from idmtools.entities.iworkflow_item import IWorkflowItem
from idmtools.entities.simulation import Simulation

class ThisAnalyzer(IAnalyzer):

    def __init__(self,):
        super().__init__(filenames=['output/InsetChart.json'])

    def map(self, data: Dict[str, Any], item: Union[IWorkflowItem, Simulation]) -> Any:
        """
        Called for Each WorkItem/Simulation.
        """
        # return data dictionary
        output = dict()

        # add tags
        output.update(item.tags)

        return data

    def reduce(self, all_data: Dict[Union[IWorkflowItem, Simulation], Any]) -> Any:

        print('In reduce length of all_data:', len(all_data))

if __name__ == "__main__":

    platform = Platform('SLURM')

    # Setup analyzers on Experiment
    analysis = PlatformAnalysis(platform=platform, experiment_ids=['166539b9-57a9-ec11-a9f5-9440c9be2c51'],
                                analyzers=[ThisAnalyzer],  analysis_name="SSMT analysis")
    analysis.analyze()

```

# Extras

System Information
====================
cwd: C:\Users\katherinero\Documents\projects\other\zero_dose_2022
data_directory: C:\Users\katherinero\.local_data
default_docket_socket_path: /var/run/docker.sock
environment_variables: {'ALLUSERSPROFILE': 'C:\\ProgramData', 'APPDATA': 'C:\\Users\\katherinero\\AppData\\Roaming', 'CCP_HOME': 'C:\\Program Files\\Microsoft HPC Pack 2012\\', 'CCP_INC': 'C:\\Program Files\\Microsoft HPC Pack 2012 SDK\\Include\\', 'CCP_JOBTEMPLATE': 'Default', 'CCP_LIB32': 'C:\\Program Files\\Microsoft HPC Pack 2012 SDK\\Lib\\i386\\', 'CCP_LIB64': 'C:\\Program Files\\Microsoft HPC Pack 2012 SDK\\Lib\\amd64\\', 'CCP_SDK': 'C:\\Program Files\\Microsoft HPC Pack 2012 SDK\\', 'COMMONPROGRAMFILES': 'C:\\Program Files\\Common Files', 'COMMONPROGRAMFILES(X86)': 'C:\\Program Files (x86)\\Common Files', 'COMMONPROGRAMW6432': 'C:\\Program Files\\Common Files', 'COMPUTERNAME': 'BMGF-JS7TP13', 'COMSPEC': 'C:\\Windows\\system32\\cmd.exe', 'CONDA_DEFAULT_ENV': 'C:\\Users\\katherinero\\Documents\\projects\\other\\zero_dose_2022\\envs', 'CONDA_EXE': 'C:\\Users\\katherinero\\Miniconda3\\condabin\\..\\Scripts\\conda.exe', 'CONDA_PREFIX': 'C:\\Users\\katherinero\\Documents\\projects\\other\\zero_dose_2022\\envs', 'CONDA_PREFIX_1': 'C:\\Users\\katherinero\\Miniconda3', 'CONDA_PROMPT_MODIFIER': "'(envs)'", 'CONDA_PYTHON_EXE': 'C:\\Users\\katherinero\\Miniconda3\\python.exe', 'CONDA_SHLVL': '2', 'DRIVERDATA': 'C:\\Windows\\System32\\Drivers\\DriverData', 'FPS_BROWSER_APP_PROFILE_STRING': 'Internet Explorer', 'FPS_BROWSER_USER_PROFILE_STRING': 'Default', 'HOMEDRIVE': 'C:', 'HOMEPATH': '\\Users\\katherinero', 'IDM_BOOST_PATH': 'C:\\Users\\katherinero\\Documents\\code\\boost_1_61_0', 'IDM_PYTHON3_PATH': 'C:\\Users\\katherinero\\AppData\\Local\\Programs\\Python\\Python36\\', 'LOCALAPPDATA': 'C:\\Users\\katherinero\\AppData\\Local', 'LOGONSERVER': '\\\\BMGF-JS7TP13', 'MSMPI_BIN': 'C:\\Program Files\\Microsoft MPI\\Bin\\', 'NUMBER_OF_PROCESSORS': '8', 'ONEDRIVE': 'C:\\Users\\katherinero\\OneDrive - Bill & Melinda Gates Foundation', 'ONEDRIVECOMMERCIAL': 'C:\\Users\\katherinero\\OneDrive - Bill & Melinda Gates Foundation', 'OS': 'Windows_NT', 'PATH': 'C:\\Users\\katherinero\\Documents\\projects\\other\\zero_dose_2022\\envs;C:\\Users\\katherinero\\Documents\\projects\\other\\zero_dose_2022\\envs\\Library\\mingw-w64\\bin;C:\\Users\\katherinero\\Documents\\projects\\other\\zero_dose_2022\\envs\\Library\\usr\\bin;C:\\Users\\katherinero\\Documents\\projects\\other\\zero_dose_2022\\envs\\Library\\bin;C:\\Users\\katherinero\\Documents\\projects\\other\\zero_dose_2022\\envs\\Scripts;C:\\Users\\katherinero\\Documents\\projects\\other\\zero_dose_2022\\envs\\bin;C:\\Users\\katherinero\\Miniconda3\\condabin;C:\\Program Files\\Microsoft MPI\\Bin;C:\\Program Files\\Microsoft HPC Pack 2012\\Bin;C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0;C:\\Windows\\System32\\OpenSSH;C:\\Windows\\system32\\config\\systemprofile\\.dnx\\bin;C:\\Program Files\\Microsoft DNX\\Dnvm;C:\\Program Files\\Microsoft SQL Server\\130\\Tools\\Binn;C:\\Program Files (x86)\\Windows Kits\\8.1\\Windows Performance Toolkit;C:\\Program Files\\Microsoft SQL Server\\120\\Tools\\Binn;C:\\Program Files\\Docker\\Docker\\resources\\bin;C:\\ProgramData\\DockerDesktop\\version-bin;C:\\Users\\katherinero\\AppData\\Local\\Programs\\Python\\Python36\\Scripts;C:\\Users\\katherinero\\AppData\\Local\\Programs\\Python\\Python36;C:\\Users\\katherinero\\Miniconda3;C:\\Users\\katherinero\\Miniconda3\\Library\\mingw-w64\\bin;C:\\Users\\katherinero\\Miniconda3\\Library\\usr\\bin;C:\\Users\\katherinero\\Miniconda3\\Library\\bin;C:\\Users\\katherinero\\Miniconda3\\Scripts;C:\\Users\\katherinero\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\katherinero\\AppData\\Local\\GitHubDesktop\\bin;C:\\Users\\katherinero\\AppData\\Local\\Programs\\Git\\cmd', 'PATHEXT': '.COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC', 'PROCESSOR_ARCHITECTURE': 'AMD64', 'PROCESSOR_IDENTIFIER': 'Intel64 Family 6 Model 126 Stepping 5, GenuineIntel', 'PROCESSOR_LEVEL': '6', 'PROCESSOR_REVISION': '7e05', 'PROGRAMDATA': 'C:\\ProgramData', 'PROGRAMFILES': 'C:\\Program Files', 'PROGRAMFILES(X86)': 'C:\\Program Files (x86)', 'PROGRAMW6432': 'C:\\Program Files', 'PROMPT': "'(envs)'$P$G", 'PSMODULEPATH': 'C:\\Program Files\\WindowsPowerShell\\Modules;C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules', 'PUBLIC': 'C:\\Users\\Public', 'RTOOLS40_HOME': 'C:\\Users\\katherinero\\Documents\\code\\rtools40', 'SESSIONNAME': 'Console', 'SYSTEMDRIVE': 'C:', 'SYSTEMROOT': 'C:\\Windows', 'TEMP': 'C:\\Users\\KATHER~1\\AppData\\Local\\Temp', 'TMP': 'C:\\Users\\KATHER~1\\AppData\\Local\\Temp', 'USERDNSDOMAIN': 'seattle.main.gatesfoundation.org', 'USERDOMAIN': 'SEATTLE', 'USERDOMAIN_ROAMINGPROFILE': 'SEATTLE', 'USERNAME': 'katherinero', 'USERPROFILE': 'C:\\Users\\katherinero', 'VS140COMNTOOLS': 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\Common7\\Tools\\', 'WINDIR': 'C:\\Windows', 'ZES_ENABLE_SYSMAN': '1', 'IDMTOOLS_NO_CONFIG_WARNING': '1'}
hostname: BMGF-JS7TP13
os_name: Windows
python_build: ('default', 'Oct 12 2021 03:01:40')
python_packages: ['argon2-cffi-bindings==21.2.0', 'argon2-cffi==21.3.0', 'arrow==1.2.2', 'astor==0.8.1', 'asttokens==2.0.5', 'astunparse==1.6.3', 'atomicwrites==1.4.0', 'attrs==21.4.0', 'backcall==0.2.0', 'backoff==1.10.0', 'beautifulsoup4==4.10.0', 'binaryornot==0.4.4', 'bleach==4.1.0', 'bs4==0.0.1', 'cached-property==1.5.2', 'certifi==2021.10.8', 'cffi==1.15.0', 'chardet==4.0.0', 'charset-normalizer==2.0.12', 'click-plugins==1.1.1', 'click==7.1.2', 'colorama==0.4.4', 'coloredlogs==15.0.1', 'cookiecutter==1.7.3', 'cycler==0.11.0', 'debugpy==1.5.1', 'decorator==5.1.1', 'defusedxml==0.7.1', 'diskcache==5.2.1', 'docopt==0.6.2', 'emod-api==1.22.10', 'emodpy==1.16.2', 'entrypoints==0.3', 'executing==0.8.3', 'fonttools==4.31.2', 'future==0.18.2', 'geographiclib==1.52', 'humanfriendly==10.0', 'idmtools-cli==1.6.4', 'idmtools-models==1.6.4', 'idmtools-platform-comps==1.6.4', 'idmtools==1.6.4', 'idna==3.3', 'importlib-metadata==4.11.3', 'iniconfig==1.1.1', 'ipykernel==6.9.1', 'ipython-genutils==0.2.0', 'ipython==8.1.1', 'jedi==0.18.1', 'jinja2-time==0.2.0', 'jinja2==3.0.3', 'joblib==1.1.0', 'jsonschema==3.2.0', 'jupyter-client==7.1.2', 'jupyter-core==4.9.2', 'jupyterlab-pygments==0.1.2', 'keyring==23.5.0', 'kiwisolver==1.4.0', 'lz4==4.0.0', 'markupsafe==2.1.1', 'matplotlib-inline==0.1.2', 'matplotlib==3.5.1', 'mistune==0.8.4', 'more-itertools==8.8.0', 'nbclient==0.5.11', 'nbconvert==6.1.0', 'nbformat==5.1.3', 'nest-asyncio==1.5.1', 'notebook==6.4.8', 'numpy==1.22.3', 'packaging==20.9', 'pandas==1.1.5', 'pandocfilters==1.5.0', 'parso==0.8.3', 'pickleshare==0.7.5', 'pillow==9.0.1', 'pip==21.2.2', 'pipreqs==0.4.11', 'pluggy==0.13.1', 'poyo==0.5.0', 'prodict==0.8.18', 'prometheus-client==0.13.1', 'prompt-toolkit==3.0.20', 'pure-eval==0.2.2', 'py==1.11.0', 'pycomps==2.5.0', 'pycparser==2.21', 'pygit2==1.5.0', 'pygments==2.11.2', 'pyparsing==3.0.7', 'pyperclip==1.8.2', 'pyproj==3.3.0', 'pyreadline3==3.4.1', 'pyrsistent==0.18.0', 'pytest-lazy-fixture==0.6.3', 'pytest==7.1.1', 'python-dateutil==2.8.2', 'python-slugify==6.1.1', 'pytz==2022.1', 'pywin32-ctypes==0.2.0', 'pywin32==302', 'pywinpty==2.0.2', 'pyyaml==5.4.1', 'pyzmq==22.3.0', 'requests==2.27.1', 'scikit-learn==1.0.2', 'scipy==1.8.0', 'send2trash==1.8.0', 'setuptools==58.0.4', 'shapely==1.8.1.post1', 'six==1.16.0', 'sklearn==0.0', 'soupsieve==2.3.1', 'stack-data==0.2.0', 'tabulate==0.8.9', 'terminado==0.13.1', 'testpath==0.5.0', 'text-unidecode==1.3', 'threadpoolctl==3.1.0', 'tomli==2.0.1', 'tornado==6.1', 'tqdm==4.63.0', 'traitlets==5.1.1', 'typing-extensions==4.1.1', 'urllib3==1.26.9', 'wcwidth==0.2.5', 'webencodings==0.5.1', 'wheel==0.37.1', 'wincertstore==0.2', 'xdg==5.1.1', 'yarg==0.1.9', 'yaspin==1.5.0', 'zipp==3.7.0']
python_version: 3.8.12
system_architecture: AMD64
system_architecture_details: ('64bit', 'WindowsPE')
system_processor: Intel64 Family 6 Model 126 Stepping 5, GenuineIntel
system_version: 10.0.19043
user: katherinero
user_group_str: 1000:1000
version: 1.6.4.0
```
